numpyro/optim.py

```diff
@@ -55,6 +60,23 @@ def update(self, g: _Params, state: _IterOptState) -> _IterOptState:
         opt_state = self.update_fn(i, g, opt_state)
         return i + 1, opt_state
 
+    def step(self, fn: Callable, state: _IterOptState) -> _IterOptState:
```

Comment:
This seems clean, but the `step` / `update` names seem a bit arbitrary. Another option could be to add `fn=None` keyword argument to `update`.

Reply:
Good idea!! I think we should go for that option.

Reply:
@neerajprad One issue with move the implementation to `update` is the output will be inconsistent. Here `step` returns both `update` output and loss value. How about renaming `step` to something else?

Reply:
Agreed, the new name seems more suitable.